src/chunkserver/datastore/file_pool.cpp

```diff
@@ -375,6 +397,125 @@ void FilePool::CleanWorker() {
+    uint64_t needSpace = (poolOpt_.filePoolSize /
+                        bytesPerPage - currentState_.chunkNum) * bytesPerPage;
+    uint64_t vaildSpace = 0;
+    curve::fs::FileSystemInfo finfo;
+    if (poolOpt_.filePoolSize / bytesPerPage < currentState_.chunkNum) {
+        LOG(WARNING) << "The number of chunks that are pre-allocated"
+                        "is less than the number of initialized chunks.";
+        LOG(INFO) << "It is no need to format chunks.";
+        formatStat_.preAllocateNum = 0;
+        formatStat_.allocateChunkNum = 0;
+        return true;
+    }
+
+    int r = fsptr_->Statfs(currentdir_, &finfo);
+    if (r != 0) {
+        LOG(ERROR) << "get disk usage info failed!";
+        return false;
+    }
+    vaildSpace = finfo.available;
+    LOG(INFO) << "free space = " << finfo.available
+            << ", total space = " << finfo.total
+            << ", need space = " << needSpace;
+    if (vaildSpace < needSpace) {
+        LOG(ERROR) << "disk free space not enough.";
+        return false;
+    }
+
+    formatStat_.preAllocateNum = needSpace / bytesPerPage;
+    formatStat_.allocateChunkNum = 0;
+    LOG(INFO) << "preAllocateNum = " << formatStat_.preAllocateNum;
+    return true;
+}
+
+bool FilePool::WaitFormatDone() {
+    std::unique_lock<std::mutex> lk(mtx_);
+    if (formatStat_.allocateChunkNum.load() != formatStat_.preAllocateNum) {
+        cond_.wait(lk, [&]() {
+            return formatStat_.allocateChunkNum.load() ==
+                   formatStat_.preAllocateNum;
+        });
+    }
+    lk.unlock();
+    if (formatThread_.joinable()) {
+        formatThread_.join();
+    }
+    return true;
+}
+
+bool FilePool::StopFormatting() {
+    if (formatAlived_.exchange(false)) {
+        LOG(INFO) << "Stop formatting...";
+        if (formatThread_.joinable()) {
+            formatThread_.join();
+        }
+        LOG(INFO) << "Stop format thread ok.";
+    }
+    return true;
+}
+
+int FilePool::FormatWorker() {
+    std::vector<Thread> threads;
+    LOG(INFO) << "format work start!";
+
+    std::atomic_bool is_wrong(false);
+    std::atomic<uint32_t> allocatIndex{0};
+    uint64_t chunksIndexOffset =
+             this->currentmaxfilenum_.fetch_add(formatStat_.preAllocateNum);
+    LOG(INFO) << "format chunk index offset: " << chunksIndexOffset;
+    auto formatTask = [&]() -> int {
+        LOG(INFO) << "format thread has been work!";
+        while (!is_wrong.load() && this->formatAlived_.load()) {
+            uint32_t chunkIndex = 0;
+            if ((chunkIndex = allocatIndex.fetch_add(1))
+                >= formatStat_.preAllocateNum) {
+                allocatIndex.fetch_sub(1);
+                break;
+            }
+            std::string chunkPath = this->currentdir_ + "/" +
+                std::to_string(chunksIndexOffset + chunkIndex) +
+                kCleanChunkSuffix_;
+            int res = this->AllocateChunk(chunkPath);
```

Comment:
Is it necessary to add some flow control to prevent formatting from taking up too much disk bandwidth? 

Or, manually start and stop the formatting task through a switch?